tests/e2e/tests/pilot/protocol_sniffing_test.go

```diff
@@ -0,0 +1,15 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilot
```

Comment:
? where is the rest of the code?

Reply:
Currently the e2e test will fail due to ci keeps failing in istio/proxy. This is the stub for e2e test. I can send another pr after istio/istio picks up the latest Envoy binary.

Reply:
This change shouldn't go in until we have updated the envoy binary and we have a non-empty test that succeeds.

Reply:
Add protocol sniffing e2e test. Waiting for picking up Envoy binary.